src/pages/index.js

```diff
@@ -21,80 +21,271 @@ import {
+    })
+    .finally(() => {
+      editAvatarPopup.setLoading(false, "Save");
+    });
 }
 
-//FORM VALIDATOR
+//calling avatar popup
 
-const addCardFormValidator = new FormValidator(settings, profileAddModal);
-addCardFormValidator.enableValidation();
+profileAvatarPeniclIcon.addEventListener("click", () => {
+  editAvatarPopup.open();
+});
 
-const editProfileFormValidator = new FormValidator(settings, profileEditModal);
-editProfileFormValidator.enableValidation();
+//--------------------------------------------------DELETE-----------------------------------------------------------------------------------
+// new delete card const
 
-//SECTION TO RENDER CARDS
+const deleteCardModalSelector = document.querySelector("#delete-confirm-modal");
 
-const cardListSection = new Section(
-  {
-    items: initialCards,
-    renderer: ({ name, link }) => {
-      const newCard = createCard({ name, link });
-      cardListSection.addItem(newCard);
-    },
-  },
-  cardList
+// delete card popup instance
+
+const deleteImagePopup = new PopupWithConfirm(
+  deleteCardModalSelector,
+  handleCardDelete
 );
+deleteImagePopup.setEventListeners();
 
-cardListSection.renderItems();
+// function for delete card
 
-// //POPUP WITH FORM
+function handleCardDelete() {
+  deleteImagePopup.setLoading(true);
+  const id = deleteImagePopup.getId();
+  api
 
-const newCardPopup = new PopupWithForm(profileAddModal, submitCard);
-newCardPopup.setEventListeners();
+    .deleteCard(id)
+    .then((res) => {
+      console.log(res);
+      deleteImagePopup.remove();
+      deleteImagePopup.close();
+    })
+    .catch((err) => {
+      console.error(err);
+    })
+    .finally(() => {
+      deleteImagePopup.setLoading(false, "Yes");
+    });
+  deleteImagePopup.open(id);
+}
+
+//delete card validator instance
+
+const deleteCardFormValidator = new FormValidator(
+  settings,
+  deleteCardModalSelector
+);
+deleteCardFormValidator.enableValidation();
+
+//-----------------------------------------------------------------CARDS--------------------------------------------------------------------
 
 const cardOpenPopup = new PopupWithImage(cardOpenModal);
 cardOpenPopup.setEventListeners();
 
-const editProfilePopup = new PopupWithForm(
-  profileEditModal,
-  handleEditFormSubmit
-);
-editProfilePopup.setEventListeners();
+function createCard(data) {
+  const newCard = new Card(
+    data,
+    userId,
+    "#card",
+    function handleCardClick(data) {
+      cardOpenPopup.open(data);
+    },
+
+    function handleDeleteClick() {
+      console.log(this);
+      deleteImagePopup.open();
+      deleteImagePopup.setId(data._id);
+      deleteImagePopup.setCard(newCard);
+    },
+
+    //card like count function
+
+    function handleCardLike(data) {
+      api
+        .changeLikeCardStatus(data._id, newCard.isLiked())
+        .then((res) => {
+          const likes = res.likes || [];
+          newCard.setLikes(likes);
+          newCard.toggleLikes();
```

Comment:
`toggleLikes` should toggle likes but inside this method you only **add** the color for the heart. So, you need to change  it to act as a toggler


```js
  toggleLikes() {
    if (this.isLiked()) {
      this._likeButton.classList.add("card__like-button_active");
    } else {
      this._likeButton.classList.remove("card__like-button_active");
    }
  }

```